### PR TITLE
More vanilla-alike block tick

### PR DESCRIPTION
**Pull request by Foghrye4**
Humanity is not ready for my progressive methods of tick optimization.

### CODE REVIEW

**Comment by Niko-sk2x** [comment]
Did you verify that it actually fixed the issue?

**Comment by Foghrye4** [comment]
Yes.

**Comment by Foghrye4** [comment]
I don't clearly understand why, though.

File: src/main/java/io/github/opencubicchunks/cubicchunks/core/world/cube/Cube.java [hunk 1]
```diff
@@ -335,25 +335,18 @@ public void tickCubeServer(BooleanSupplier tryToTickFaster, Random rand) {
...
         tickCubeCommon(tryToTickFaster);
 
-        Iterator<NextTickListEntry> pti = pendingTickListEntriesTreeSet.iterator();
-        while (pti.hasNext()) {
+        while (!pendingTickListEntriesTreeSet.isEmpty()) {
             assert storage != null : "There are blocks to tick in cube but getStorage() is empty!";
-            NextTickListEntry ntle = pti.next();
+            NextTickListEntry ntle = pendingTickListEntriesTreeSet.first();
             if (ntle.scheduledTime > world.getTotalWorldTime())
                 return;
```
Comment (Niko-sk2x) [inline src/main/java/io/github/opencubicchunks/cubicchunks/core/world/cube/Cube.java]: This is why it didn't work before. If this is reached, no entries will be moved from hashset to treeset.

Comment (Foghrye4) [reply src/main/java/io/github/opencubicchunks/cubicchunks/core/world/cube/Cube.java]: Ow. Should I just replace `return` with `break` then? Where was a reason why I add this `256` ticks limit (probably).

**Comment by Niko-sk2x** [comment]
The reason for this limit was that vanilla has a global limit. 

Andi really think this should be redone more vanilla way, with global ticking. The way this is done now is very likely to cause some subtle differences in some redstone circuits.

**Comment by Foghrye4** [comment]
As you wish.

**Comment by Foghrye4** [comment]
Done. It seems an only purpose of those injections was optimization.

**Comment by Niko-sk2x** [comment]
The reason for this was a very important optimization that has to be done differently. 

When the world is first created and spawn point is generated, world has hundreds of thousands scheduled gravel updates. Triggering world save before these updates are processed (which can be done by just keeping the game minimized while generating chunks) will cause Minecraft to save all the chunks together with their scheduled ticks. 

Now for chunks to save it's own ticks, it has to filter the whole list of schedules ticks. And this has to be done a few tens of thousands of times, on average with half of the initial amount of scheduled ticks. In some cases it can take up to a few minutes to save the world in this case. 

In vanilla it's less of an issue because there are 16 times less chunks. So even if it takes half a minute it's still not totally unreasonable.

**Comment by Niko-sk2x** [comment]
Also I think there are more changes needed to remove this. I don't even know how it compiles now. There is chunk saving code that also interacts with this.

**Comment by Foghrye4** [comment]
I'll take a look in vanilla code to learn what you mean. Do you want me to add some kind of temporary solution in this PR (replace `return` with `break`) for now on?

**Comment by Niko-sk2x** [comment]
Yes.

**Comment by Foghrye4** [comment]
Done and verified to work.

**Pull request merged**